### PR TITLE
Added support the `[cargo-new]` table 

### PR DESCRIPTION
**Pull request by ranger-ross**
Hey there! :wave: 

I also added support for the [`cargo-new`](https://doc.rust-lang.org/cargo/reference/config.html#cargo-new) table which only has 1 none deprecated property. 
I did not add the deprecated fields.

### CODE REVIEW

File: src/de.rs [hunk 1]
```diff
@@ -443,6 +448,64 @@ pub struct FutureIncompatReportConfig {
...
+#[non_exhaustive]
+pub struct CargoNewConfig {
+    /// Specifies the source control system to use for initializing a new repository.
+    /// Valid values are git, hg (for Mercurial), pijul, fossil or none to disable this behavior.
+    /// Defaults to git, or none if already inside a VCS repository. Can be overridden with the --vcs CLI option.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub vcs: Option<Value<VersionControlSoftware>>,
+}
+
+#[allow(clippy::exhaustive_enums)]
```
Comment (taiki-e) [inline src/de.rs:465]: I think this enum should use `#[non_exhaustive]`.

File: tests/test.rs [hunk 2]
```diff
@@ -72,7 +72,7 @@ fn assert_reference_example(de: fn(&Path, ResolveOptions) -> Result<Config>) ->
     assert_eq!(config.future_incompat_report.frequency, Some(Frequency::Always));
 
     // TODO
-    // [cargo-new]
+    assert_eq!(config.cargo_new.vcs, None);
```
Comment (taiki-e) [inline tests/test.rs:75]: We can test this by uncommenting the following two lines:

https://github.com/taiki-e/cargo-config2/blob/bb273c0e03ae844238beabbe712682d7366954b6/tests/fixtures/reference/.cargo/config.toml#L41-L42

**Review by taiki-e — CHANGES_REQUESTED** [review]
Thanks!

File: src/de.rs (hunk 1 above)
Comment (ranger-ross) [reply src/de.rs]: Thanks, good catch. I made this `#[non_exhaustive]` and reexported it at the crate root to match the other enums :+1:

File: tests/fixtures/reference/.cargo/config.toml [hunk 3]
```diff
@@ -38,8 +38,8 @@ ENV_VAR_NAME_3 = { value = "relative/path", relative = true }
 [future-incompat-report]
 frequency = 'always' # when to display a notification about a future incompat report
 
-# [cargo-new]
-# vcs = "none"              # VCS to use ('git', 'hg', 'pijul', 'fossil', 'none')
+[cargo-new]
+vcs = "none"              # VCS to use ('git', 'hg', 'pijul', 'fossil', 'none')
```
Comment (taiki-e) [inline tests/fixtures/reference/.cargo/config.toml:42]:
```suggestion
vcs = "none" # VCS to use ('git', 'hg', 'pijul', 'fossil', 'none')
```

(fix ci error: https://github.com/taiki-e/cargo-config2/actions/runs/10650634616/job/29522247723?pr=21#step:10:54)

**Review by taiki-e — APPROVED** [review]

**Comment by taiki-e** [comment]
Published in 0.1.28.

**Pull request merged**